executor/src/witgen/jit/processor.rs

```diff
@@ -170,25 +164,18 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
         &self,
         can_process: CanProcess,
         witgen: &mut WitgenInference<'a, T, FixedEval>,
-        complete: &mut HashSet<(u64, i32)>,
     ) {
-        let mut progress = true;
-        while progress {
-            progress = false;
-
-            // TODO At this point, we should call a function on `witgen`
-            // to propagate known concrete values across the identities
-            // to other known (but not concrete) variables.
-
-            for (id, row_offset) in &self.identities {
-                if complete.contains(&(id.id(), *row_offset)) {
-                    continue;
-                }
-                let result = witgen.process_identity(can_process.clone(), id, *row_offset);
-                progress |= result.progress;
-                if result.complete {
-                    complete.insert((id.id(), *row_offset));
-                }
+        loop {
+            let progress = self
+                .identities
+                .iter()
+                .map(|(id, row_offset)| {
+                    witgen.process_identity(can_process.clone(), id, *row_offset)
+                })
+                .reduce(std::ops::BitOr::bitor)
+                .unwrap_or(false);
```

Comment:
```suggestion
                .any(|(id, row_offset)| {
                    witgen.process_identity(can_process.clone(), id, *row_offset)
                });
```

This is what you mentioned in the call, right? Sounds good to me! Basically implements #2323.

Reply:
It does require a change of tests where we assert a particular code, but looking at the diff, I think the code is actually structured more logically, because there is a closer relation to the order in which things where written in PIL.